File: EDAII x Java/Arboles/MenuArbolBinario.java
import java.util.Scanner;
import java.io.IOException;

public class MenuArbolBinario{

	public static void main(String[] args){
		ArbolBinario arbol = new ArbolBinario();
		Scanner sc = new Scanner(System.in);
		int op=0,valor;

		while (op != 8){
			System.out.println("\n------ Menú Árbol Binario de Búsqueda Balanceado ------");
			System.out.println("1) Añadir nodo");
			System.out.println("2) Eliminar nodo");
			System.out.println("3) Buscar nodo");
			System.out.println("4) Altura árbol");
			System.out.println("5) Imprimir árbol");
			System.out.println("6) Limpiar árbol");
			System.out.println("7) Limpiar pantalla");
			System.out.println("8) Salir");
			System.out.print("Seleccione una opción: ");
			op = sc.nextInt();

			switch(op){
				case 1:
					System.out.println("\n------ Añadir Nodo ------");
					System.out.print("Ingrese el valor para el nuevo nodo: ");
					valor = sc.nextInt();
					arbol.añadir(new Nodo(valor));
					break;
				case 2:
					System.out.println("\n------ Eliminar Nodo ------");
					System.out.print("Ingrese el valor del nodo a eliminar: ");
					valor = sc.nextInt();
					arbol.eliminar(valor);
					break;
				case 3:
					System.out.println("\n------ Buscar Nodo ------");
					System.out.print("Ingrese el valor del nodo a buscar: ");
					valor = sc.nextInt();
					Nodo nodo = arbol.obtenerNodo(arbol.raiz,valor);
					if (nodo == null){
						System.out.println("Nodo no encontrado.");
					} else {
						System.out.println("Nodo encontrado.");
					}
					break;
				case 4:
					System.out.println("\n------ Altura Árbol ------");
					if (arbol.raiz == null){
						System.out.println("Altura: -1");
					} else {
						int altura = arbol.altura(arbol.raiz);
						System.out.println("Altura: " + altura);
					}
					break;
				case 5:
					System.out.println("\n------ Imprimir Árbol ------");
					if (arbol.raiz != null){
						System.out.print("Recorrido Prefijo: ");
						arbol.prefijo(arbol.raiz);
						System.out.print("\nRecorrido Infijo: ");
						arbol.infijo(arbol.raiz);
						System.out.print("\nRecorrido Postfijo: ");
						arbol.postfijo(arbol.raiz);
						System.out.print("\nRecorrido BFS: ");
						arbol.BFS();
					} else {
						System.out.println("Árbol vacío.");
					}
					break;
				case 6:
					System.out.println("\n------ Limpiar Árbol ------");
					arbol = new ArbolBinario();
					System.out.println("Árbol limpiado.");
					break;
				case 7:
					// Comando que limpia la pantalla de la Terminal
					System.out.print("\033[H\033[2J");
    				System.out.flush();
					break;
				case 8:
					System.out.println("\n------ Salir ------\n");
					break;
				default:
					System.out.println("Opción no válida.");
					break;
			}
		}
	}
}

File: EDAII x Python/Arboles/MenuArbolBinario.py
from ArbolBinario import ArbolBinario
from Nodo import Nodo
import os
import platform

class MenuArbolBinario:

	def main(self):
		'Constructor de la clase'
		arbol = ArbolBinario(None);
		op = 0
		valor = 0

		while op != 8:
			print("\n------ Menú Árbol Binario de Búsqueda Balanceado ------")
			print("1) Añadir nodo")
			print("2) Eliminar nodo")
			print("3) Buscar nodo")
			print("4) Altura árbol")
			print("5) Imprimir árbol")
			print("6) Limpiar árbol")
			print("7) Limpiar pantalla")
			print("8) Salir")
			op = int(input("Seleccione una opción: "))

			if op == 1:
				print("\n------ Añadir Nodo ------")
				valor = int(input("Ingrese el valor para el nuevo nodo: "))
				arbol.añadir(Nodo(valor,None,None,None))
			elif op == 2:
				print("\n------ Eliminar Nodo ------")
				valor = int(input("Ingrese el valor del nodo a eliminar: "))
				arbol.eliminar(valor)
			elif op == 3:
				print("\n------ Buscar Nodo ------")
				valor = int(input("Ingrese el valor del nodo a buscar: "))
				nodo = arbol.obtenerNodo(arbol.raiz, valor)
				if nodo == None:
					print("Nodo no encontrado")
				else:
					print("Nodo encontrado")
			elif op == 4:
				print("\n------ Altura Árbol ------")
				if arbol.raiz == None:
					print("Altura: -1")
				else:
					altura = arbol.altura(arbol.raiz)
					print("Altura: ",altura)
			elif op == 5:
				print("\n------ Imprimir Árbol ------")
				if arbol.raiz != None:
					print("Recorrido Prefijo: ")
					arbol.prefijo(arbol.raiz)
					print("Recorrido Infijo: ")
					arbol.infijo(arbol.raiz)
					print("Recorrido Postfijo: ")
					arbol.postfijo(arbol.raiz)
					print("Recorrido BFS: ")
					arbol.BFS()
				else:
					print("Árbol vacío")
			elif op == 6:
				print("\n------ Limpiar Árbol ------")
				arbol = ArbolBinario(None)
				print("Árbol limpiado.")
			elif op == 7:
				if platform.system() == 'Windows':
					uv = os.system("cls")
				else:
					uv = os.system("clear")	# MacOS / Linux
			elif op == 8:
				print("\n------ Salir ------\n")
			else:
				print("Opción no válida")

# Ejecutando el objeto
programa = MenuArbolBinario()
programa.main()